mirgecom/filter.py

```diff
@@ -23,24 +23,138 @@
 """
 import math
 import numpy as np
+import loopy as lp
+#from pytools import memoize_method
+from pytools.obj_array import obj_array_vectorized_n_args
 
 
-def get_spectral_filter(dim, order, cutoff, filter_order):
+def exponential_mode_response_function(mode_idx, alpha, cutoff, nfilt, filter_order):
+    return np.exp(-1.0 * alpha * ((sum(mode_idx) - cutoff) / nfilt)
+                  ** (2*filter_order))
+
+
+def make_spectral_filter(dim, order, cutoff, mode_response_function):
+    r"""
+    Exponential spectral filter from JSH/TW Nodal DG Methods, pp. 130, 186
+    """
+    npt = 1
+    for d in range(1, dim+1):
+        npt *= (order + d)
+    npt /= math.factorial(int(dim))
+    npt = int(npt)
```

Comment:
`len(mode_ids)` should give the same thing.

Reply:
Just need modepy issue cleared up (see above).